### ISSUE TITLE
[std] StringTools missing variants

### ISSUE THREAD

**Issue by eliasku** [HaxeFoundation/intellij-haxe#417]
Methods are missed for haxe std StringTools extension.
StringTools is very basic extension, and uncomfortable to use with current Intellij-Haxe version.
Methods are not available because `one-lined` style is used for conditional compilation.

Repro:

```
using StringTools;
class StdTodo {
    function parseStringTools() {
        var s = "";
        s.replace("3", " "); // Unresolved symbol, no completion
        StringTools.replace(s, "3", " ");// unresolved methods, no completion
    }
}
```

Why broken:
![image](https://cloud.githubusercontent.com/assets/3038174/13070595/cd706908-d49e-11e5-992d-f66d72992e10.png)

I agreed that one-line style for conditional compilation is very unreadable style, but Std library and a lot of external libraries are using it intensively.

Version: 0.9.9, IDEA 15, OSX
Related: one-line preprocessor, conditional compilation

**Comment by EBatTiVo** [comment]
Yes.  There are other bugs for this: #115, #29, #121.  @as3boyan has tried to fix it in the past, and there is commented out code in the parser that tries to deal with it, though I think he never got it working right.

The trouble is that the elements can appear anywhere within a source file, so can't be inserted into the BNF.  We had to settle for finding them at the beginning of the line.  Even then, there are many issues around refactoring, find usage, etc. where we need to parse all of the paths and verify their syntax independently.

The simple (and typical) method of just treating the non-current path as comments does not account for those usage patterns.

**Comment by mayakwd** [comment]
Already fixed

**Comment by EricBishton** [comment]
Fixed for 0.11.0 release.
